### PR TITLE
Set version artificially for the test template runs

### PR DESCRIPTION
**Pull request by chidozieononiwu**
This allows template (Test) run to go all the way to release with artificially generated versions.
Sample run https://dev.azure.com/azure-sdk/internal/_build/results?buildId=559264&view=results

### CODE REVIEW

File: eng/scripts/SetTestPipelineVersion.ps1 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+$semVarsSorted = [AzureEngSemanticVersion]::SortVersionStrings($semVars)
+LogDebug "Last Published Version $($semVarsSorted[0])"
+
+$newVersion = [AzureEngSemanticVersion]::ParseVersionString($semVarsSorted[0])
+$newVersion.IncrementAndSetToPrerelease()
+LogDebug "Version to publish [ $($newVersion.ToString()) ]"
+
+&"${PSScriptRoot}/Update-PkgVersion.ps1" -ServiceDirectory "template" `
+-PackageName 'Azure.Template' -PackageDirName "Azure.Template" -NewVersionString $newVersion.ToString() `
```
Comment (weshaggard) [inline eng/scripts/SetTestPipelineVersion.ps1:22]: Similar to the other PR's I'd like to see if we can use the change log operations for adding the notes.

File: eng/scripts/SetTestPipelineVersion.ps1 [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Overides the project file and CHANGELOG.md for the template project using the next publishable version
+# This is to help with testing the release pipeline.
+. "${PSScriptRoot}\..\common\scripts\common.ps1"
+$latestTags = git tag -l "Azure.Template*beta*"
```
Comment (weshaggard) [inline eng/scripts/SetTestPipelineVersion.ps1:4]: I'd suggest just `Azure.Template_*`.

**Review by weshaggard — APPROVED** [review]

File: eng/scripts/SetTestPipelineVersion.ps1 (hunk 2 above)
Comment (chidozieononiwu) [reply eng/scripts/SetTestPipelineVersion.ps1:4]: I'll try that. It should work as long as the sorting in semvar can handle preview and beta.

**Pull request merged**